src/cleanfigure.m

```diff
@@ -292,40 +292,42 @@ function pruneOutsideBox(meta, handle)
 end
 % =========================================================================
 function out = segmentVisible(data, dataIsInBox, xLim, yLim)
-    % Given a bounding box {x,y}Lim, loop through all pairs of subsequent nodes
-    % in p and determine whether the line between the pair crosses the box.
+    % Given a bounding box {x,y}Lim, determine whether the line between all 
+    % pairs of subsequent data points crosses the box.
     n = size(data, 1);
     out = false(n-1, 1);
+    
+    % Only check if there is more than 1 point    
+    if n>1
+        % One of the neighbors is inside the box and the other is finite
+        nextVisible = (dataIsInBox(2:end)   & all(isfinite(data(1:end-1)), 2));
+        thisVisible = (dataIsInBox(1:end-1) & all(isfinite(data(2:end)), 2));
```

Comment:
Just a minor thing, maybe replace this with
```matlab
idx = 1:n-1;
thisVisible = (dataIsInBox(idx)   & all(isfinite(data(idx+1)), 2));
nextVisible = (dataIsInBox(idx+1) & all(isfinite(data(idx  )), 2));
```

Main points: `this` before `next` and columnwise alignment of the statements. I care less about factoring out `idx` (both are equally clear, and I don't know whether one or the other is faster/more memory efficient in practice).